utils/config/nodes.hpp

```diff
@@ -254,40 +208,38 @@ class base_set_node : public leaf_node {
 
     base_set_node(void);
 
+    /// Checks whether the node has been set by the user.
+    ///
+    /// \return True if a value has been set in the node.
     bool is_set(void) const;
 
     /// Gets the value stored in the node.
     ///
-    /// \todo Figure out why Doxygen is unable to pick up the documentation for
-    /// this function from the nodes.ipp file.
-    ///
     /// \pre The node must have a value.
     ///
     /// \return The value in the node.
     const value_type& value(void) const;
 
     /// Gets the read-write value stored in the node.
     ///
-    /// \todo Figure out why Doxygen is unable to pick up the documentation for
-    /// this function from the nodes.ipp file.
-    ///
     /// \pre The node must have a value.
     ///
     /// \return The value in the node.
     value_type& value(void);
 
     /// Sets the value of the node.
-    ///
-    /// \todo Figure out why Doxygen is unable to pick up the documentation for
-    /// this function from the nodes.ipp file.
-    ///
-    /// \param value_ The new value to set the node to.
     void set(const value_type&);
 
+    /// Sets the value of the node from a raw string representation.
```

Comment:
How come this doesn't have to document the parameter?

Oh wait, because we are not naming it. Maybe this docstring should actually be in the implementation file, where the parameter is actually named and used? Otherwise, why isn't Doxygen complaining about an undocumented parameter?

Reply:
> How come this doesn't have to document the parameter?
> 
> Oh wait, because we are not naming it.

Bingo!

> Maybe this docstring should actually be in the implementation file, where the parameter is actually named and used?

It is :). From utils/config/nodes.ipp:

```
233 /// Sets the value of the node from a raw string representation.
234 ///
235 /// \param raw_value The value to set the node to.
236 ///
237 /// \throw value_error If the value is invalid.
238 template< typename ValueType >
239 void
240 config::native_leaf_node< ValueType >::set_string(const std::string& raw_value)
```

The reason why I added these method summaries is explained in the new commit message:

```
utils/config/nodes.hpp:

Remove doxygen docs which duplicated properties already documented in
`utils/config/nodes.[ci]pp`. Also, remove parameters to mute warnings
about undocumented parameters in downstream children of classes that
inherited the parent class methods and their properties, but lacked
their own implementations, and provide a minimal amount of documentation
so downstream consumers which have not overridden the methods are documented.
```

> Otherwise, why isn't Doxygen complaining about an undocumented parameter?

🤷‍♀️, but it might be because they're documented where they're implemented. It's just annoying because some of the downstream implementations don't actually implement the methods and that's what's triggering the warnings from Doxygen.